cc_library/src/scripts/controlboard.py

```diff
@@ -87,40 +87,41 @@ def get_status(self):
         status += "}"
         return status
 
+    # Todo: make the library check for this instruction and coll it directly
     def perform_instruction(self, contents):
         """
         Set here the mapping from messages to methods.
         Should return warning when illegal instruction was sent
         or instruction could not be performed.
         """
-        instruction = contents.get("instruction")
-        if instruction == "test":
-            self.test()
-        elif instruction == "blink":
-            self.blink(contents)
-        elif instruction == "turnOff":
-            self.turn_off(contents)
-        elif instruction == "turnOn":
-            self.turn_on(contents)
-        else:
-            return True
-        return None
-
-    def blink(self, data):
-        led = getattr(self, data.get("led"))
-        interval = data.get("interval")
+        for action in contents:
```

Comment:
i don't think you should receive multiple instructions at once

Reply:
why not? In case of handling multiple LEDs at one you do want this in one instruction in order to be able to perform this without noticeable delay between the LEDs